rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -2279,3 +2279,153 @@ rocalToDecibels(
     }
     return output;
 }
+
+RocalTensor ROCAL_API_CALL
+rocalResample(RocalContext p_context,
+              RocalTensor p_input,
+              RocalTensor p_input_resample_rate,
+              RocalTensorOutputType rocal_tensor_output_datatype,
+              bool is_output,
+              float sample_hint) {
+    if (!p_context || !p_input_resample_rate || !p_input)
+        THROW("Null values passed as input")
+    Tensor* resampled_output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    auto input_resample_rate = static_cast<Tensor*>(p_input_resample_rate);
+    RocalTensorDataType op_tensor_data_type;
+    try {
+        TensorInfo output_info = input->info();
+        op_tensor_data_type = (RocalTensorDataType)rocal_tensor_output_datatype;
+        output_info.set_tensor_layout(RocalTensorlayout::NONE);
+        output_info.set_data_type(op_tensor_data_type);
+        if (sample_hint > 0) {
+            std::vector<size_t> max_dims = output_info.max_shape();
+            std::vector<size_t> dims = output_info.dims();
+            dims[1] = std::ceil(sample_hint);
+            dims[2] = max_dims[1];
+            output_info.set_dims(dims);
+        } else {
+            THROW("Please pass a valid resample hint")
+        }
+        resampled_output = context->master_graph->create_tensor(output_info, is_output);
+        resampled_output->reset_tensor_roi();
+        context->master_graph->add_node<ResampleNode>({input}, {resampled_output})->init(input_resample_rate, 50.0);
+    } catch (const std::exception& e) {
+        context->capture_error(e.what());
+        ERR(e.what())
+    }
+    return resampled_output;
+}
+
+RocalTensor rocalTensorMulScalar(RocalContext p_context,
+                                 RocalTensor p_input,
+                                 bool is_output,
+                                 RocalTensorOutputType rocal_tensor_output_type,
+                                 float scalar) {
+    if (!p_context || !p_input)
+        THROW("Null values passed as input")
+    Tensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    RocalTensorDataType op_tensor_data_type;
+    try {
+        op_tensor_data_type = (RocalTensorDataType)rocal_tensor_output_type;
```

Comment:
Do we support all data types for this function, If not check and add a condition

Please use static cast